### PR TITLE
FWE Networking

### PR DESCRIPTION
**Pull request by domesticmouse**
_Description of what this PR is changing or adding, and why:_

_Issues fixed by this PR (if any):_

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/get-started/fwe/networking.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+title: Networking and data
+description: Learn how to network your Flutter app.
+prev:
+  title: Handling user input
+next:
+    title: Local data and caching
+---
+
+While it is said that no man is an island, a Flutter app without any networking capability can feel a tad disconnected.
```
Comment (sfshaza2) [inline src/get-started/fwe/networking.md:10]:
```suggestion
While it's said that "no man is an island",
a Flutter app without any networking capability can feel a tad disconnected.
```

File: src/get-started/fwe/networking.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+In this section you will learn how to add networking features to your Flutter app.
+Your app will retrieve data, parse JSON into usable in memory representations, and then send data out again.
```
Comment (sfshaza2) [inline src/get-started/fwe/networking.md:12]:
```suggestion
This page covers how to add networking features to your Flutter app.
Your app will retrieve data, parse JSON into usable in memory representations,
and then send data out again.
```

File: src/get-started/fwe/networking.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+## Introduction to retrieving data over the network
+
+The following two tutorials introduce you to the [`http`][] package which enables your app to make [HTTP][] requests with ease, whether you are running on Android, iOS, inside a web browser, or natively on Windows, macOS or Linux. 
+The first tutorial shows you how to make an unauthenticated `GET` request to a website, parse the retrieved data as `JSON` and then display the resulting data.
+The second tutorial builds on the first by adding authentication headers, enabling access to web servers requiring authorization. The article by the Mozilla Developer Network (MDN) gives more background on how authorisation works on the web.
```
Comment (sfshaza2) [inline src/get-started/fwe/networking.md:18]: OMG, you hate line breaks! Also, our docs use American spelling, not Ozzie spelling.  :D

A question. Do these tutorials work for mobile apps when accessing websites?

```suggestion
The following two tutorials introduce you to the
[`http`][] package, which enables your app to make
[HTTP][] requests with ease, whether you are running on Android,
iOS, inside a web browser, or natively on Windows, macOS, or Linux. 
The first tutorial shows you how to make an unauthenticated `GET` request
to a website, parse the retrieved data as `JSON` and then display the resulting data.
The second tutorial builds on the first by adding authentication headers,
enabling access to web servers requiring authorization.
The article by the Mozilla Developer Network (MDN)
gives more background on how authorization works on the web.
```

File: src/get-started/fwe/networking.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+* Article: [MDN's article on Authorization for websites][]
+
+## Making data retrived from the network useful
+
+Once you are retrieving data from the network, you will want a way to convert the data from the network into something that you can easily work with in Dart.
+The tutorials in the previous section used hand rolled Dart to convert network data into an in memory representation. 
+In this section you'll see other options for handling this conversion.
+The first is a YouTube video giving an overview of the [`freezed` package][]. 
+While the second is a codelab covering patterns and records using the case study of parsing JSON. 
+On point, no?
```
Comment (sfshaza2) [inline src/get-started/fwe/networking.md:31]:
```suggestion
Once you retrieve data from the network,
you need a way to convert the data from the network
into something that you can easily work with in Dart.
The tutorials in the previous section used hand rolled Dart
to convert network data into an in-memory representation. 
In this section, you'll see other options for handling this conversion.
The first links to a YouTube video showing an overview of the [`freezed` package][]. 
The second links to a codelab that covers patterns and records
using a case study of parsing JSON. 
```

File: src/get-started/fwe/networking.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+* Codelab: [Dive into Dart's patterns and records][]
+
+## Going both ways, getting data out again
+
+Now that you have mastered the art of retrieving data, it's time to look at pushing data out.
+This starts with sending data to the network, but then dives into asynchronicity.
+The truth is, once you are in a conversation over the network, you'll need to deal with the fact that web servers that are far away take a while to respond, and you can't stop rendering to the screen while you wait for packets to round trip.
+Dart has really great support for asynchonicity, as does Flutter.
+You'll learn all about Dart's support in a tutorial, then see Flutter's capability covered in a Widget of the Week video.
+Once you are through that, you'll learn how to debug network traffic using DevTool's Network View.
```
Comment (sfshaza2) [inline src/get-started/fwe/networking.md:43]: I really like this section though, yes, I do have some comments.

```suggestion
Now that you've mastered the art of retrieving data, it's time to look at pushing data out.
This information starts with sending data to the network, but then dives into asynchronicity.
The truth is, once you are in a conversation over the network,
you'll need to deal with the fact that web servers that are physically far away
can take a while to respond, and you can't stop rendering to the screen
while you wait for packets to round trip.
Dart has really great support for asynchronicity, as does Flutter.
You'll learn all about Dart's support in a tutorial,
then see Flutter's capability covered in a Widget of the Week video.
Once you complete that, you'll learn how to debug
network traffic using DevTool's Network View.
```

File: src/get-started/fwe/networking.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+* Tutorial: [Asynchronous programming: futures, async, await][]
+* YouTube video: [FutureBuilder (Widget of the Week)][]
+* Article: [Using the Network View][]
+
+## Extension material
+
+Now that you have mastered using Flutter's networking APIs, it'll help to see Flutter's network usage in context.
+The first is a codelab ostensibly on creating Adaptive Apps in Flutter, but it uses a web server written in Dart to work around the web browsers' [Cross-Origin Resource Sharing (CORS) restrictions][].
+Next is a long form YouTube video where Flutter DevRel alumni Fitz talks about how the location of data matters for Flutter apps.
+Finally, a really useful series of articles by Flutter GDE Anna (Domashych) Leushchenko covering advanced networking in Flutter.
```
Comment (sfshaza2) [inline src/get-started/fwe/networking.md:56]:
```suggestion
Now that you've mastered using Flutter's networking APIs,
it helps to see Flutter's network usage in context.
The first codelab (ostensibly on creating Adaptive Apps in Flutter),
uses a web server written in Dart to work around the web browsers'
[Cross-Origin Resource Sharing (CORS) restrictions][].
Next, a long-form YouTube video where Flutter DevRel alumnus,
Fitz, talks about how the location of data matters for Flutter apps.
Finally, a really useful series of articles by Flutter GDE
Anna (Domashych) Leushchenko covering advanced networking in Flutter.
```

File: src/get-started/fwe/networking.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+[Fetch data from the internet]: https://docs.flutter.dev/cookbook/networking/fetch-data
+[Freezed (Package of the Week)]: https://www.youtube.com/watch?v=RaThk0fiphA
+[`freezed` package]: https://pub.dev/packages/freezed
+[FutureBuilder (Widget of the Week)]: https://www.youtube.com/watch?v=zEdw_1B7JHY
+[`http`]: https://pub.dev/packages/http
+[HTTP]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Overview
+[Keeping it local: Managing a Flutter app's data]: https://www.youtube.com/watch?v=uCbHxLA9t9E
+[Make authenticated requests]: https://docs.flutter.dev/cookbook/networking/authenticated-requests
+[MDN's article on Authorization for websites]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization
+[Using the Network View]: https://docs.flutter.dev/tools/devtools/network
```
Comment (sfshaza2) [inline src/get-started/fwe/networking.md:77]: Look at that beautiful link list in alphabetical order!

```suggestion
[Adaptive Apps in Flutter]: {{site.codelabs}}/codelabs/flutter-adaptive-app
[Asynchronous programming: futures, async, await]: {{site.dart-site}}/codelabs/async-await
[Basic and advanced networking in Dart and Flutter]: {{site.medium}}/tide-engineering-team/basic-and-advanced-networking-in-dart-and-flutter-the-tide-way-part-0-introduction-33ac040a4a1c
[Cross-Origin Resource Sharing (CORS) restrictions]: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
[Dive into Dart's patterns and records]: {{site.codelabs}}/codelabs/dart-patterns-records
[Fetch data from the internet]: {{site.url}}/cookbook/networking/fetch-data
[Freezed (Package of the Week)]: {{site.youtube-site}}/watch?v=RaThk0fiphA
[`freezed` package]: {{site.pub-pkg}}/freezed
[FutureBuilder (Widget of the Week)]: {{site.youtube-site}}/watch?v=zEdw_1B7JHY
[`http`]: {{site.pub-pkg}}/http
[HTTP]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Overview
[Keeping it local: Managing a Flutter app's data]: {{site.youtube-site}}/watch?v=uCbHxLA9t9E
[Make authenticated requests]: {{site.url}/cookbook/networking/authenticated-requests
[MDN's article on Authorization for websites]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization
[Using the Network View]: https://docs.flutter.dev/tools/devtools/network
```

**Review by sfshaza2 — COMMENTED** [review]
Other than your allergy to line breaks, I really like this page, @domesticmouse!

File: src/get-started/fwe/networking.md (hunk 3 above)
Comment (domesticmouse) [reply src/get-started/fwe/networking.md:20]: Yes, I'm Australian. Yes, I love run on sentences. I'm trying to get better, but it still slips through =)

As to whether these tutorials are useful from a mobile, I'd have to ask the question as to who is running Xcode and their Flutter tooling on a phone...

**Comment by domesticmouse** [comment]
> Other than your allergy to line breaks, I really like this page, @domesticmouse!

I was attempting to follow [sembr](https://sembr.org/). Sounds like I was doing it wrong =)

**Comment by sfshaza2** [comment]
Interesting. I've never heard of sembr. Their website says, "When writing text with a compatible markup language, add a line break after each substantial unit of thought."  See, the fact that we are ALSO limited by 80 chars, means that we add line breaks after _unsubstantial_ units of thought.  :D

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Pull request merged**